src/js/utils/DSLParserUtil.js

```diff
@@ -0,0 +1,174 @@
+import DSLFilterTypes from '../constants/DSLFilterTypes';
+import DSLCombinerTypes from '../constants/DSLCombinerTypes';
+import {FilterNode, CombinerNode} from '../structs/DSLASTNodes';
+
+/**
+ * Template function for combining two filters, used by the parser
+ *
+ * When this function is 'instantiated', the first two arguments are bound by
+ * the parser to the child filter functions (see below) and the `this` context
+ * to an instance of a `CombinerNode` representing the token AST node.
+ *
+ * When this function is used, the user has to supply the final two arguments.
+ * The `filters` object is a library with the available filters that can be used
+ * in this context, while the `resultset` is an instance of `List` or `Tree` on
+ * which the filers have to be applied.
+ *
+ * @this CombinerNode
+ * @param {function} [filter1] - The first child filter function (auto-bound)
+ * @param {function} [filter2] - The second child filter function (auto-bound)
+ * @param {Object} filters - An object containing the valid filters that can be used
+ * @param {List} resultset - An instance of List or Tree containing the items to filter
+ *
+ * @returns {List} - Returns the filtered resultset
+ */
+function combine_fn(filter1, filter2, filters, resultset) {
```

Comment:
Is it a requirement of JISON to use snake_case_notation? Would be awesome if we could just use mixedCaseNotation.